server/src/modules/enquiry/enquiry.controller.ts

```diff
@@ -0,0 +1,61 @@
+import { createLogger } from '../../bootstrap/logging'
+import { ControllerHandler } from '../../types/response-handler'
+import { MailService } from '../mail/mail.service'
+import { Enquiry } from '../../types/mail-type'
+import { AgencyService } from '../agency/agency.service'
+
+const logger = createLogger(module)
+
+export class EnquiryController {
+  constructor(
+    private mailService: Public<MailService>,
+    private agencyService: Public<AgencyService>,
+  ) {}
+
+  postEnquiry: ControllerHandler<
+    unknown,
+    unknown,
+    {
+      agencyID: Array<string>
+      enquiry: Enquiry
+    }
+  > = async (req, res) => {
+    try {
+      const agencyEmail = await Promise.all(
+        req.body.agencyID.map(async (id) => {
+          const agency = await this.agencyService.findOneById(parseInt(id))
+          return agency ? agency.email : ''
+        }),
+      )
+      const ccEmail = ['enquiries@ask.gov.sg']
+
+      if (agencyEmail.length === 0) {
+        agencyEmail.push('enquiries@ask.gov.sg')
+        ccEmail.pop()
+      }
+      const validData = agencyEmail.every((email) => email !== '')
+
+      if (!validData) {
+        return res
+          .status(404)
+          .json({ message: 'One or more agency IDs are invalid' })
+      }
+
+      const enquiry = req.body.enquiry
+      enquiry.questionTitle = 'AskGov: ' + enquiry.questionTitle
+
+      await this.mailService.sendEnquiry({ agencyEmail, ccEmail, enquiry })
```

Comment:
There is quite a fair bit of biz logic that goes on here. Bear in mind also that this will change in later iterations, so we probably do not want to be messing around with the controller then.

I think it would be prudent if all this is done within an EnquiryService that in turn delegates to:
- Either AgencyService or the Sequelize model for Agency to look up the agency (the latter preferred), and;
- MailService, to send the mail.

I realise that this breaks with conventional wisdom that it's okay for the controller to pass to multiple services, but I'm mindful that how enquiries are sent in may change over time.

Reply:
Noted, I added EnquiryService to have MailService and use the sequalize model.